server/api.go

```diff
@@ -147,14 +148,19 @@ func (a *API) processLifecycle(w http.ResponseWriter, req *http.Request) {
 
 	a.p.API.LogDebug("Lifecycle activity request", "activities", lifecycleEvents)
 
+	errors := ""
 	for _, event := range lifecycleEvents.Value {
 		if event.ClientState != a.p.getConfiguration().WebhookSecret {
-			a.p.API.LogError("Invalid webhook secret received in lifecycle event")
+			a.p.metricsService.ObserveLifecycleEvent(event.LifecycleEvent, metrics.DiscardedReasonInvalidWebhookSecret)
+			errors += "Invalid webhook secret"
```

Comment:
This errors handling strategy is currently copy pasted from the change event handler. I think we can probably improve on this, but opted not to touch that logic except to copy it here. Notably, we now return `http.StatusBadRequest` when this occurs vs. treating it as a valid event. Thoughts?

Reply:
According to Microsoft's documentation, we should invalidate a change notification in this case, as it means that the change notification may have originated somewhere else. See this - https://learn.microsoft.com/en-us/graph/change-notifications-delivery-webhooks?tabs=http#processing-the-change-notification

However, it's also possible that the client state received in the change notification is correct but a system administrator has changed the webhook secret in the plugin's configuration. Right now, we don't have any logic that deletes the existing subscriptions and recreates them with the new webhook secret whenever it changes. Also, we're not storing the previous values of the webhook secret anywhere. So, right now, there's no way to validate if the client state received in the change notification is one of the webhook secrets previously configured for the plugin.

So, to sum up, we have to respond with `BadRequest` as the change notification can be from somewhere else and we can't risk security. But, we have to come up with a plan to confirm the source of the change notification and how to properly validate the client state.